controllers/podchaos/types.go

```diff
@@ -83,19 +85,15 @@ func (r *Reconciler) schedulePodChaos(podchaos *v1alpha1.PodChaos, req ctrl.Requ
 		}
 		return reconciler.Reconcile(req)
 	case v1alpha1.PodFailureAction:
-		if podchaos.Spec.Duration == nil {
-			r.Log.Error(nil, "schedule podchaos should define duration", "action", podchaos.Spec.Action)
-			return ctrl.Result{}, nil
-		}
-		r := twophasePodfailure.NewTwoPhaseReconciler(r.Client, r.Log.WithValues("action", "pod-failure"), req)
-		reconciler := twophase.NewReconciler(r, r.Client, r.Log)
-		return reconciler.Reconcile(req)
+		tr = podfailure.NewTwoPhaseReconciler(r.Client, r.Log.WithValues("action", "pod-kill"), req)
 	default:
-		return r.defaultResponse(podchaos), nil
+		return r.invalidActionResponse(podchaos), nil
 	}
+	reconciler := twophase.NewReconciler(tr, tr.Client, tr.Log)
```

Comment:
why unbox the `twophase.Reconciler` instance and box it again here?

Reply:
updated.